dts/bindings/i2c/ti,tca9546a.yaml

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2020, Innoseis BV
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  TCA9546a I2C switch
+
+  Each channel is represented by a separate device tree child node
+  with a "ti,tca9546a,channel" compatible.
+  The channel node can then be used as a standard i2c bus controller
+  like in the following simplified device tree:
+
+      /* The tca9546a node must be a child of an i2c controller */
+      mux: tca9546a@77 {
+          compatible = "ti,tca9546a";
+          reg = <0x77>;
+
+          mux_i2c@0 {
+              compatible = "ti,tca9546a,channel";
+              reg = <0>;
+
+              temp_sens_0: tmp116@49 {
+                  compatible = "ti,tmp116";
+                  reg = <0x49>;
+              };
+          };
+
+          mux_i2c@1 {
+              compatible = "ti,tca9546a,channel";
+              reg = <1>;
+
+              temp_sens_1: tmp116@49 {
+                  compatible = "ti,tmp116";
+                  reg = <0x49>;
+              };
+          };
+      };
+
+compatible: "ti,tca9546a"
+
+include: [i2c-device.yaml]
+
+child-binding:
+  description: TCA9546a I2C switch channel node
+  compatible: "ti,tca9546a,channel"
```

Comment:
Sorry, but I just noticed this. I think this should be `ti,tca9546a-channel` instead. Usually, `-` is used to separate words in the `device` portion of a `vnd,device` compatible.